fs-index/src/index.rs

```diff
@@ -1,1120 +1,501 @@
+    ///   root path of the index).
+    ///
+    /// # Returns
+    /// Returns `Ok(resource)` if the resource was successfully removed from the
+    /// index.
+    ///
+    /// # Errors
+    /// - If the file still exists in the file system.
+    /// - If the resource does not exist in the index.
+    pub fn track_removal<P: AsRef<Path>>(
+        &mut self,
+        relative_path: P,
+    ) -> Result<IndexedResource<Id>> {
+        log::debug!("Tracking removal of file: {:?}", relative_path.as_ref());
+
+        let path = relative_path.as_ref();
+        let full_path = self.root.join(path);
+        if full_path.exists() {
+            return Err(ArklibError::Path(format!(
+                "File still exists: {:?}",
+                full_path
+            )));
+        }
 
-        assert_ne!(new1, new2);
-        assert_ne!(new1, old1);
+        // Remove the resource from the index
+        let resource = self
+            .path_to_resource
+            .remove(path)
+            .ok_or_else(|| anyhow!("Resource not found: {}", path.display()))?;
+
+        // Remove the resource from the id_to_resources map
+        if let Some(resources) = self.id_to_resources.get_mut(&resource.id) {
+            resources.retain(|r| r.path != resource.path);
+            if resources.is_empty() {
+                self.id_to_resources.remove(&resource.id);
+            }
+        }
 
-        assert!(new1 > old1);
-        assert!(new1 > old2);
-        assert!(new2 > old1);
-        assert!(new2 > old2);
-        assert!(new2 > new1);
+        Ok(resource)
     }
 
-    /// Test the performance of `ResourceIndex::build` on a specific directory.
+    /// Track the modification of a file in the resource index.
     ///
-    /// This test evaluates the performance of building a resource
-    /// index using the `ResourceIndex::build` method on a given directory.
-    /// It measures the time taken to build the resource index and prints the
-    /// number of collisions detected.
-    #[test]
-    fn test_build_resource_index() {
-        use std::time::Instant;
-
-        let path = "../test-assets/"; // The path to the directory to index
-        assert!(
-            std::path::Path::new(path).is_dir(),
-            "The provided path is not a directory or does not exist"
+    /// This method checks if the file exists in the file system and removes the
+    /// old resource from the index before adding the new resource to the
+    /// index.
+    ///
+    /// # Arguments
+    /// * `relative_path` - The relative path of the file to be modified.
+    ///
+    /// # Returns
+    /// Returns `Ok(new_resource)` if the resource was successfully modified in
+    /// the index.
+    ///
+    /// # Errors
+    /// - If there was a problem removing the old resource from the index.
+    /// - If there was a problem adding the new resource to the index.
+    pub fn track_modification<P: AsRef<Path>>(
```

Comment:
Original motivation was here:
- https://github.com/ARK-Builders/arklib/issues/38

The use-case was updating the resource in one app, after its modification in another app.

But now I think, this function is not needed, because:
1. The editor app can have different flow, like opening a resource by one path, and saving the modified version to another path. Reactive API is much more robust here, especially with `notify`. We should ensure that `update_all` is efficient, too. We had a bug in there causing very slow re-indexing every time a single update was made. That's why the `track_update` appeared originally.
2. If the editor overwrites the resource by same path, then it still can use combination of `track_deletion` and `track_addition`.